eng/common/scripts/release-docs-metadata.ps1

```diff
@@ -0,0 +1,208 @@
+      Write-Host "Unable to retrieve service metadata for packageId $($pkgInfo.PackageId)"
+    }
+
+    $headerContentMatch = (Select-String -InputObject $pkgInfo.ReadmeContent -Pattern 'Azure .+? (client|plugin|shared) library for (JavaScript|Java|Python|\.NET|C)').Matches[0]
+
+    if ($headerContentMatch){
+      $header = "---`r`ntitle: $headerContentMatch`r`nkeywords: Azure, $lang, SDK, API, $service $($pkgInfo.PackageId)`r`nauthor: maggiepint`r`nms.author: magpint`r`nms.date: $date`r`nms.topic: article`r`nms.prod: azure`r`nms.technology: azure`r`nms.devlang: $lang`r`nms.service: $service`r`n---`r`n"
+      $fileContent = $pkgInfo.ReadmeContent -replace $headerContentMatch, "$headerContentMatch - Version $($pkgInfo.PackageVersion) `r`n"
+      return "$header $fileContent"
+    }
+    else {
+      return ""
+    }
+}
+
+$apiUrl = "https://api.github.com/repos/$repoId"
+$pushUrl = "https://$($env:GH_TOKEN)@github.com/$TargetDocRepo.git"
+$branchScriptLocation = (Join-Path $PSScriptRoot git-branch-push.ps1)
+$prScriptLocation = (Join-Path $PSScriptRoot Submit-PullRequest.ps1)
+$DocRepoOwner = $TargetDocRepo.split("/")[0]
+$DocRepoName = $TargetDocRepo.split("/")[1]
+
+$pkgs = VerifyPackages -pkgRepository $Repository `
+  -artifactLocation $ArtifactLocation `
+  -workingDirectory $WorkDirectory `
+  -apiUrl $apiUrl `
+  -releaseSha $ReleaseSHA `
+  -exitOnError $False
+
+if ($pkgs) {
+  Write-Host "Given the visible artifacts, readmes will be copied for the following packages"
+  Write-Host ($pkgs | % { $_.PackageId }) 
+
+  foreach ($packageInfo in $pkgs) {
+    # ensure the working directory is clean for the doc repo clone
+    Remove-Item -Recurse -Force "$WorkDirectory/*"
+    
+    # sync the doc repo
+    $semVer = [AzureEngSemanticVersion]::ParseVersionString($packageInfo.PackageVersion)
+    $rdSuffix = ""
+    $targetDocBranch = $LatestBranch
+    if ($semVer.IsPreRelease) {
+      $targetDocBranch = $PreviewBranch
+      $rdSuffix = "-pre"
+    }
+    $prBranch = "$targetDocBranch-rdme"
+    $repoLocation = SyncDocRepo -docRepo $TargetDocRepo -workingDirectory $WorkDirectory -sourceBranch $targetDocBranch -prBranch $prBranch
+
+    Write-Host "Selected Branch is $targetDocBranch, PRBranch is $prBranch"
+
+    $readmeName = "$($packageInfo.PackageId.Replace('azure-','').Replace('Azure.', '').Replace('@azure/', '').ToLower())-readme$rdSuffix.md"
+    $readmeLocation = Join-Path $repoLocation $DocRepoContentLocation $readmeName
+    $adjustedContent = GetAdjustedReadmeContent -pkgInfo $packageInfo -lang $Language
+
+    if ($adjustedContent) {
+      try {
+        Push-Location $repoLocation
+        Set-Content -Path $readmeLocation -Value $adjustedContent -Force
+
+        Write-Host "git add -A ."
+        git add -A .
+
+        # commit the changes to the PR Branch. If others have pushed to the target branch we want to rebase off it and commit our changes regardless
+        & $branchScriptLocation -PRBranchName $prBranch `
```

Comment:
I'd suggest you only have this script do the source updates and move the clone, branching, and pull-request work out into the yml pipeline that will be doing this work.

Reply:
Hmm. I don't really agree, but I'll try and see how it looks.

Reply:
It is generally a good idea to have the script be single purposed and if you can imagine yourself wanting to do this from the command  line without all the orchestration of cloning or creating pull requests then it seems like it would be a good idea to have them be separate steps.